examples/hybrid/gravitywave_parameterization/gravity_wave_parameterization.jl

```diff
@@ -67,12 +129,32 @@ function gravity_wave_tendency!(Yₜ, Y, p, t)
     # .     may be calculated
     #
 
-    # source level: get the index of the level that is closest to the source height (GFDL uses the fist level below instead)
-    source_level = argmin(
-        abs.(
-            unique(parent(Fields.coordinate_field(Y.c).z) .- gw_source_height)
-        ),
-    )
+    # source level: get the index of the level that is closest to the source height/pressure (GFDL uses the fist level below instead)
+    if config == "column"
+        source_level = similar(Fields.level(Y.c.ρ, 1))
+        Fields.bycolumn(axes(ᶜρ)) do colidx
+            parent(source_level[colidx]) .= argmin(
+                abs.(
+                    parent(Fields.coordinate_field(Y.c).z[colidx]) .-
+                    gw_source_height
+                ),
+            )[1]
+        end
+    elseif config == "sphere"
+        (; ᶜK, ᶜp) = p
+        thermo_params = CAP.thermodynamics_params(params)
+        ᶜuₕ = Y.c.uₕ
+        ᶠw = Y.f.w
+        @. ᶜK = norm_sqr(C123(ᶜuₕ) + C123(ᶜinterp(ᶠw))) / 2
+        thermo_state!(ᶜts, Y, thermo_params, p.thermo_dispatcher, ᶜinterp, ᶜK)
+        @. ᶜp = TD.air_pressure(thermo_params, ᶜts)
+
+        source_level = similar(Fields.level(Y.c.ρ, 1))
+        Fields.bycolumn(axes(ᶜρ)) do colidx
+            parent(source_level[colidx]) .=
+                argmin(abs.(parent(ᶜp[colidx]) .- gw_source_pressure))[1]
+        end
+    end
```

Comment:
In the spirit of Charlie's comments above, would it make sense to put these into separate functions and use multiple dispatch?